tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java

```diff
@@ -1308,6 +1332,74 @@ private void sendKavaPlayRequest(String entryId) {
         NetworkUtils.sendKavaAnalytics(context, partnerId, entryId, NetworkUtils.KAVA_EVENT_PLAY_REQUEST, sessionId);
     }
 
+    private PKAdvertisingController getPkAdvertisingController() {
+        if (pkAdvertisingController == null) {
+            pkAdvertisingController = new PKAdvertisingController();
+        }
+        return pkAdvertisingController;
+    }
+
+    /**
+     * Advertising Configuration with JSON
+     * Set null in case if App does not want to configure anything
+     * for the next media playback (ChangeMedia)
+     *
+     * @param advertisingJson AdvertisingConfig JSON
+     */
+    public void setAdvertisingConfig(@Nullable String advertisingJson) {
+        log.d("setAdvertisingConfig Json");
+        if (TextUtils.isEmpty(advertisingJson)) {
+            log.d("Advertising config is empty. Hence clearing the current advertising config.");
+            this.advertisingConfig = null;
+            return;
+        }
+
+        if (this.advertisingConfig != null) {
+            this.advertisingConfig = null;
+        }
+
+        String imaPlugin = KnownPlugin.ima.name();
+        if (initOptions.pluginConfigs != null && initOptions.pluginConfigs.hasConfig(imaPlugin)) {
+            AdvertisingConfig advertisingConfig = new Gson().fromJson(advertisingJson, AdvertisingConfig.class);
+            if (advertisingConfig != null) {
+                this.advertisingConfig = advertisingConfig;
+            } else {
+                log.e("Malformed AdvertisingConfig Json");
+            }
+        } else {
+            log.w("IMAPlugin needs to be configured in order to use Advertising feature. \n " +
+                    "You can pass empty adtag url while configuring IMAPlugin");
+        }
+    }
+
+    /**
+     * Advertising Configuration with AdvertisingConfig object
+     * Set null in case if App does not want to configure anything
+     * for the next media playback (ChangeMedia)
+     *
+     * @param advertisingConfig AdvertisingConfig object
+     */
+    public void setAdvertisingConfig(@Nullable AdvertisingConfig advertisingConfig) {
+        log.d("setAdvertisingConfig object");
+        if (advertisingConfig == null) {
+            log.d("Advertising config is null. Hence clearing the current advertising config.");
+            this.advertisingConfig = null;
+            return;
+        }
+
+        if (this.advertisingConfig != null) {
+            this.advertisingConfig = null;
+        }
+
+        String imaPlugin = KnownPlugin.ima.name();
+        if (initOptions.pluginConfigs != null && initOptions.pluginConfigs.hasConfig(imaPlugin)) {
+            this.advertisingConfig = advertisingConfig;
+        } else {
+            log.w("IMAPlugin needs to be configured in order to use Advertising feature. \n " +
+                    "You can pass empty adtag url while configuring IMAPlugin");
+        }
+    }
+
```

Comment:
lets make the public API short and call internal unified logic for both cases

Reply:
Done. Moved it in a single method receiving object.